### PR TITLE
Develop to master

### PR DESCRIPTION
**Pull request by ThrawnCA**
- Add more allowed file extensions.
- Allow archives to specify any format so long as they're well formed, since it's legitimate for the resource format to refer to the files inside the archive.

### CODE REVIEW

**Review by duttonw — APPROVED** [review]

**Comment by ThrawnCA** [comment]
@duttonw Can you test https://servicesmadesimpler.govnet.qld.gov.au/jira/browse/QOL-7762 before we merge this?

**Comment by duttonw** [comment]
Hi @ThrawnCA , have added comment to our ticket system, But will comment here also. Zip's do allow all formats per our internal spec. This is a major improvement to our security and reduction in work required in manual intervention after the fact when items are loaded with incorrect type picked. 

In the future, we can raise with ckan core as a request to do auto mine type guess picking as default if its a file upload or manual choice as well as verify its on the whitelist in. This might be in the core or a standalone plugin based on the work you did in splitting the function into its own discreet part.

**Comment by ThrawnCA** [comment]
> In the future, we can raise with ckan core as a request to do auto mine type guess picking as default if its a file upload or manual choice as well as verify its on the whitelist in. This might be in the core or a standalone plugin based on the work you did in splitting the function into its own discreet part.

CKAN core developers initially weren't keen, stating that it's a hard problem and pointing out the archive format thing, which is why I've developed it this way. I've updated them now and am awaiting an email response.

I've also opened an issue with `ckanext-security` (https://github.com/data-govt-nz/ckanext-security/issues/41), since it seems relevant to them.

**Pull request merged**
